### DISCUSSION TITLE
Nested form inside React Portal submits parent form too?

### DISCUSSION THREAD

**Discussion by Mudbill** [react-hook-form/react-hook-form#9891] [Q&A] [answered]
I have a form which is somewhat large and spans a few components.

At some point deeper down, I want a dialog/modal to show with a separate form, where the result will affect the overarching form.

The dialog is using a React Portal, which places it outside the first \<form> element in the DOM. Yet still, when I submit this dialog form, the first form is also submitted. Why?

See example:
https://codesandbox.io/s/distracted-scooby-omvmi9?file=/src/App.tsx

(I know that HTML5 disallows nested form elements, but the portal should avoid that, so this seems to be caused by hook form)

**Comment by Sajarin-M** [comment, marked as answer]
Yes, the events are still propagated even if the child form is inside a portal, as stated in the [React documentation](https://beta.reactjs.org/reference/react-dom/createPortal#caveats)
If you really need to nest the form, try stopping the propagation in the child form like this:
```js
const onSubmit = (e) => {
  e.stopPropagation();
  form.handleSubmit((values) => alert("Submitted form 2"))(e);
};
```

**Reply by Mudbill** [reply]
I guess I figured the event propagation was part of the native DOM and wouldn't care about the order React logically tracks things in.
This should do the trick though.
